src/syntax/specification/specification.ts

```diff
@@ -51,37 +54,49 @@ export function registerElementSpecificationEntry(
         | IElementSpecificationExpression
         | IElementSpecificationStatement
         | IElementSpecificationBlock;
+
+    return name in _elementSpecification;
 }
 
 /**
  * Registers a syntax element specification from a given specification entry table.
  * @param specification specification entry table object with key-value pairs of element name and
  *  corresponding specification entry data
+ * @returns a list of boolean , false if element name already exists else true
  */
 export function registerElementSpecificationEntries(specification: {
     [identifier: string]: IElementSpecification;
-}): void {
+}): boolean[] {
+    const registerElementsEntries: boolean[] = [];
     Object.entries(specification).forEach(([identifier, specification]) =>
-        registerElementSpecificationEntry(identifier, specification)
+        registerElementsEntries.push(registerElementSpecificationEntry(identifier, specification))
     );
+    return registerElementsEntries;
 }
 
 /**
  * Removes specification for a syntax element.
  * @param name name of the syntax element
+ * @returns true if element is successfully removed, else false
  */
-export function removeElementSpecificationEntry(name: string): void {
+export function removeElementSpecificationEntry(name: string): boolean {
     if (name in _elementSpecification) {
         delete _elementSpecification[name];
+        return !(name in _elementSpecification);
+    } else {
+        return true;
     }
 }
 
 /**
  * Removes specification for a list of syntax element.
  * @param names list of names of the syntax element
+ * @returns a list of boolean, true if element is successfully removed, else false
```

Comment:
a list of boolean values, \`true\` if element is successfully removed else \`false\`